scripts/scraper-ng/rules/html-require-recipe-ingredients/ingredient-handlers/data-examples.js

```diff
@@ -27,6 +28,7 @@ function checkExample(exampleNodes, logger) {
           "EmbedLiveSample must be final node",
           "nodes-after-live-sample"
         );
+        ok = false;
```

Comment:
Right. I wasn't sure if a page can be said to "have" a malformed ingredient. Since we check the well-formedness of ingredients first, I figured it was easier to say that a page either has a well-formed ingredient, or it doesn't have the ingredient, and that's that.

To make this a little more concrete, I was imagining a scenario like putting the wrong heading on a section (e.g., an authors puts a compat macro underneath a specifications heading). Should I be generous and assume that the author intended to include a specifications section and check the order based on that assumption? I'm not sure, as an author, that I _want_ a permissive linter—especially if I meant to have a compat section there.

Also, I thought it was an easier API to enforce and reason about: if you don't log any errors, then you must return a position node and if you do log any errors, then you must not.

Reply:
Yes, I think that's fine. As you say, it gives us a clear definition for when a node should be returned versus `null`.